Tasks/Common/ServiceFabricHelpers/Strings/resources.resjson/en-US/resources.resjson

```diff
@@ -8,5 +8,6 @@
   "loc.messages.ErrorOnCertificateImport": "An error occurred attempting to import the certificate. Ensure that your service endpoint is configured properly with a correct certificate value and, if the certificate is password-protected, a valid password. Error message: {0}",
   "loc.messages.ImportedCertificate": "Imported cluster client certificate with thumbprint '{0}'.",
   "loc.messages.ServiceFabricSDKNotInstalled": "Service Fabric SDK is not installed on the build agent.  Go to https://aka.ms/servicefabric to download and install it.",
-  "loc.messages.UseWindowsSecurity": "Windows security is specified. -WindowsCredential switch will be used while connecting."
+  "loc.messages.UseWindowsSecurity": "Windows security is specified. -WindowsCredential switch will be used while connecting.",
+  "loc.messages.NonCertBasedAuthForTextEncryptionError": "In order to encrypt the data for this task, the cluster being used must have certificate-based authentication."
```

Comment:
I've updated the logic to lookup the thumbprint of the ServerCertThumbprint.  If it exists, then use that thumbprint, otherwise throw an error with a message of "The cluster's server certificate with thumbprint '{0}' is required in order to encrypt text but the certificate could not be found on the agent machine."

---
In reply to: [136201695](https://github.com/Microsoft/vsts-tasks/pull/5076#discussion_r136201695) [](ancestors = 136201695)